### PR TITLE
Issue #1335: auto: --until モードの Round 内 issue 処理順序を Value/Size と関連性で最適化

### LINKED ISSUE #1335: auto: --until モードの Round 内 issue 処理順序を Value/Size と関連性で最適化
## Background

`/auto --batch --until <query>` (#953) の Until mode は、`scripts/resolve-batch-query.sh` が返す順序 (昇順 issue 番号、機械的ソート) のまま `ROUND_LIST` を List mode に渡して処理する。既存の blocked-by gate (List mode Step 4) はハードな依存関係のみを扱い、それ以外の issue 間の関係は一切考慮されない。

2026-08-10 の `--until "label:theme/observability"` 実運用検証 (10 issue) で、この処理順序に改善余地があることを示す具体例が観察された:

- **#1049** (`run-auto-sub: --write-manual-recovery の manual_intervention event under-counting 調査`) と **#1153** (`auto: --write-manual-recovery に task notification 文言を記録し harness-stop / external-signal を判別`) は同じ `--write-manual-recovery` を対象にしており、#1153 が先に landing すれば #1049 の調査 AC が自然に解決しやすくなる可能性が高い。ただし両者は title prefix が異なる (`run-auto-sub:` / `auto:`) ため、単純な文字列一致によるクラスタリングでは検出できない
- **#1105** (`detect-wrapper-anomaly: review の silent no-op を独立パターンとして切り出し`) と **#1323** (`detect-wrapper-anomaly: background task 完了通知待ちによる silent no-op を Tier 2 signature に追加`) は同一 title prefix (`detect-wrapper-anomaly:`) を共有しており、同じモジュール/パターンカタログを触る。こちらは文字列一致で検出可能

auto/verify のフルパイプラインは issue あたり数十分〜規模次第で時間単位のコストがかかるため、それに比べれば Round 開始時に処理順序を判断するコスト (機械的ソートや軽量な AI 推論) は無視できる規模である。

## Purpose

Until mode の Round 内処理順序を、関連する issue 同士が近接して処理されるよう、また ROI (Value 高 / Size 小) の高い issue から着手されるよう最適化し、手戻り・二重編集・コンフリクトを減らす。

## Proposal (Outline)

Until mode の **Step 3 (`resolve-batch-query.sh` による ROUND_LIST 解決) の直後、Step 5 で List mode 処理を開始する前**に、`ROUND_LIST` の並び替えステップを挿入する。

3 種類のシグナルを組み合わせる:

1. **Value/Size ソート (機械的・AI 判断不要)**: Projects V2 に既に存在する Value/Size フィールド値をそのまま使い、Value 高 × Size 小の issue を優先する (cost-of-delay 的な優先度付け)
2. **title prefix クラスタリング (軽量・文字列一致)**: issue title の prefix (例: `detect-wrapper-anomaly:`) が一致するものを隣接させる。#1105/#1323 のようなケースをコストゼロで検出できる
3. **意味的関係判断 (AI 推論)**: ROUND_LIST 内の issue body を読み、prefix が異なるため 2 では拾えない関係 (#1049/#1153 のようなケース) を検出し、順序に反映する。パイプライン全体のコストと比較して許容範囲と判断

既存の blocked-by gate (List mode Step 4) はハードな依存関係のブロックを担うもので、本提案の並び替えはそれを代替しない (並び替えは「両方処理可能な issue 同士の望ましい順序」を扱い、ブロック判定そのものには関与しない)。

3 シグナルの優先順位・組み合わせ方法 (例: Value/Size を第一キー、クラスタリングをタイブレーカーにするか、あるいはクラスタリングを優先するか) は `/spec` での判断が必要。

### Pre-merge

- [x] Until mode の Round ループに ROUND_LIST の並び替えステップが実装されている  
- [x] Value/Size ソート・title prefix クラスタリング・意味的関係判断の3シグナルの優先順位/組み合わせ方が明記されている  
- [x] 既存の blocked-by gate (List mode Step 4) との関係が明文化されている (並び替えは blocked-by gate を代替しない)  
- [x] 実運用でこの並び替えが意図通り機能する (関連 issue が隣接処理される、または高 ROI issue が先に処理される) ことを確認する手順が定義されている  

### Post-merge

なし

## Related

- #953 (`--until` モード本体)
- #1334 (Until mode の別の改善: retrospective 起票 issue の Theme label 未達 = chicken-and-egg 問題。本 Issue とは独立に着地可能)
- 発生元: 2026-08-10 セッションでの `--until "label:theme/observability"` 実運用検証 (#1049/#1153, #1105/#1323 の具体例を観察)

### PR DESCRIPTION
**Pull request by saito**
## Summary
- `scripts/compute-round-order.sh` を新規追加: `ROUND_LIST` 内の各 issue について title/Size/Value を 1 issue あたり `gh-graphql.sh` 1 回のラウンドトリップで取得し (Project field 優先、`size/*`/`value/*` ラベルフォールバック)、ROI (Value/Size) を計算する
- `modules/round-ordering.md` を新規追加: ROI・title prefix クラスタリング・意味的関係判断 (LLM) の3シグナルを cluster-first で組み合わせて `ROUND_LIST` を並び替える手順、既存の blocked-by gate との関係、実運用確認手順を定義
- `skills/auto/SKILL.md` の Until mode (`--batch --until <query>`) step 6 に並び替えステップを挿入 (`ROUND_LIST` 記録直後・`write_batch` 呼び出し直前)。`allowed-tools` に `compute-round-order.sh` を追加
- `docs/structure.md` / `docs/workflow.md` およびその `docs/ja/` ミラーを同期

## Verification (pre-merge)
- `bats --jobs 18 tests/` フルスイート 1702 件 全 PASS (skills/auto/SKILL.md の behavioral change 検出により full-suite override)
- `python3 scripts/validate-skill-syntax.py skills/` 0 error / 0 warning
- `bash scripts/check-forbidden-expressions.sh` 0 件
- `bash scripts/check-allowed-tools.sh skills/` 0 件
- `bash scripts/check-translation-sync.sh` docs/structure.md・docs/workflow.md ともに IN_SYNC
- Issue #1335 の pre-merge AC 4件、verify command anchor (`round-ordering.md` / `cluster-first` / `Relationship to the blocked-by gate` / `Confirming real-world effectiveness`) すべて確認しチェック済み

## Verification (post-merge)
- なし (Issue 本文の Post-merge セクションは空)

closes #1335

### CODE REVIEW

File: modules/round-ordering.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+2. For each Issue, extract a title prefix: the text before the first `:` in the title, only if
+   that `:` occurs within the first 40 characters of the title. Issues with no such `:` have no
+   prefix (each becomes its own singleton candidate cluster).
+3. Group Issues sharing an identical, non-empty prefix into the same cluster. Issues with no
+   prefix, or a prefix unique to them within `ROUND_LIST`, remain singleton clusters at this
+   stage.
+
+### Signal 3 (semantic, LLM judgment)
+
+4. For each singleton cluster from step 3, read the Issue body of every other Issue still in
```
Comment (saito) [inline modules/round-ordering.md]: [Documentation] CONSIDER: Step 4 of "Signal 3 (semantic, LLM judgment)" says to read "the Issue body of every other Issue still in `ROUND_LIST`" but does not explicitly say the singleton's own body must also be read for the comparison to be meaningful. Likely implied, but could be misread as requiring only the *other* issues' bodies.

Suggested fix: reword to "...read the Issue body of the singleton itself and of every other Issue still in `ROUND_LIST`, and look for..." to remove the ambiguity.

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| Until mode の Round ループに ROUND_LIST の並び替えステップが実装されている | PASS | `skills/auto/SKILL.md` Until mode step 6 で `round-ordering.md` を `ROUND_LIST` 記録直後・`write_batch` 呼び出し直前に呼び出している (section_contains + rubric ともに確認) |
| Value/Size ソート・title prefix クラスタリング・意味的関係判断の3シグナルの優先順位/組み合わせ方が明記されている | PASS | `modules/round-ordering.md` に `cluster-first` の定義と Signal 1-3 の組み合わせ方が明記されている |
| 既存の blocked-by gate (List mode Step 4) との関係が明文化されている | PASS | `modules/round-ordering.md` § "Relationship to the blocked-by gate" に明記されている |
| 実運用でこの並び替えが意図通り機能することを確認する手順が定義されている | PASS | `modules/round-ordering.md` § "Confirming real-world effectiveness" に具体的な確認手順が定義されている |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| Language Convention check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### General Comments (issues where path/line cannot be identified)

**[Edge Case/Robustness] tests/compute-round-order.bats**
`scripts/compute-round-order.sh` の2つのフォールバック経路がテストされていない: (1) ラベルフォールバック成功パス (GraphQL が空を返しラベルから SIZE/VALUE が解決される経路。現状は「GraphQL 空・一致ラベルなし」の neutral-fallback ケースのみテストされている)、(2) title フォールバックパス (`gh issue view --json title` が呼ばれる、GraphQL の title が空/null の場合の経路)。Severity: SHOULD

Recommended fix:
`tests/compute-round-order.bats` に2ケース追加: (a) mock `gh` が `size/M`/`value/3` ラベルを返し GraphQL フィールドが空のケース (ラベルから SIZE/VALUE が採用されることを確認)、(b) GraphQL の `title` が null で `gh issue view --json title` フォールバックの mock がタイトルを返すケース (5列目にタイトルが現れることを確認)。

### Issue Count Summary

- Posted as line comments: 1 (MUST: 0, SHOULD: 0, CONSIDER: 1)
- General Comments (in body): 1 (SHOULD: 1)

---
Next: run `/merge 1338`.

**Comment by saito** [comment]
Review: [Edge Case/Robustness] tests/compute-round-order.bats — scripts/compute-round-order.sh の label-fallback 成功パスと title-fallback パスが未テスト。Severity: SHOULD

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response (Lightweight Integrated Review — all 4 aspects)

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Edge Case/Robustness | tests/compute-round-order.bats | — | label-fallback / title-fallback パスが未テスト (SHOULD) | Resolved | 2件のテストケースを追加 (label fallback: size/M・value/3 のラベルから解決、title fallback: gh issue view の title フォールバック) |
| Documentation | modules/round-ordering.md | 40 | Signal 3 step 4 の記述が singleton 自身の body 読み込みを明示していない (CONSIDER) | Resolved | 文言を「the singleton itself and of every other Issue」に修正 |

### Lightweight Re-check Results

- Tests: PASS (`bats tests/compute-round-order.bats` 8/8、新規2件含む)
- `python3 scripts/validate-skill-syntax.py skills/`: PASS (0 error / 0 warning)
- Fix diff check: PASS
- New issues: none
